apps/server/src/api/v2/file-upload.ts

```diff
@@ -14,3 +21,17 @@ function handleFileUpload(ctx) {
     },
   };
 }
+
+function delPrevFilesIfAny(currentFileName) {
+  const uploadsDir = config.uploadsPath;
+  return getFileNames(uploadsDir).then(files => {
+    const resourceId = currentFileName.split('-')[0];
+    const filesToDel = files.filter(
+      file => file.substr(0, file.indexOf('-')) === resourceId && file !== currentFileName
+    );
+    filesToDel.map(async fileToDel => {
```

Comment:
I'm confused about this line, are you trying to make the function wait for all the files to be deleted before returning? if so then there should be a return statement of the promises, otherwise this function won't wait.

Another thing is that `async/await` doesn't work with array methods, i.e. even if you use async/await inside the `map()` call it won't make it wait for the one iteration before going to next one, all is synchronous. If you want to wait for multiple promises then you need to use `Promise.all()`:

```
const deleteFile = promisify(fs.unlink);
return Promise.all(
  filesToDel.map(fileTodel => {
    const filePathToDel = path.join(uploadsDir, fileToDel);
    return fs.unlink(filePathToDel);
  })
);
```